torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -213,14 +212,19 @@ Tensor & copy_(Tensor & self, const Tensor & src, bool non_blocking) {
   }
   {
     at::AutoNonVariableTypeMode non_var_type_mode(true);
-    self_.copy_(src_, non_blocking);
+    static auto op = c10::Dispatcher::singleton()
+      .findSchemaOrThrow("aten::copy_", "")
+      .typed<Tensor & (Tensor &, const Tensor &, bool)>();
+    c10::Dispatcher::singleton()
+      .redispatch<Tensor &, Tensor &, const Tensor &, bool>(op, ks & c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::AutogradOther), self_, src_, non_blocking);
```

Comment:
one way to cut down on the size of this might be to stick `c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::AutogradOther)` in a `constexpr` in `DispatchKeySet` or wherever and use it even from codegen - like, `PostAutogradDispatchKeys` or something would actually be semantically meaningful in the same way as `AutoNonVariableTypeMode` (but clearer :)

Reply:
Just pushed a commit that tried to address most of the duplication around the PR- but I'll try to address this in a followup